addons/website_sale/models/product.py

```diff
@@ -141,11 +141,11 @@ class ProductTemplate(models.Model):
     website_public_price = fields.Float('Website public price', compute='_website_price', digits=dp.get_precision('Product Price'))
 
     def _website_price(self):
-        self.mapped('product_variant_id')
+        products = self.env['product.product'].browse(self.mapped('product_variant_id').ids)
 
-        for p in self:
-            p.website_price = p.product_variant_id.website_price
-            p.website_public_price = p.product_variant_id.website_public_price
+        for p, p2 in zip(self, products):
+            p.website_price = p2.website_price
+            p.website_public_price = p2.website_public_price
```

Comment:
The issue is how the prefetching data is fed with all the product variants. You can avoid this in a simpler way:

    for template, product in zip(self.with_prefetch(), self.mapped('product_variant_id')):
        template.website_price = product.website_price
        template.website_public_price = product.website_public_price

The call `self.mapped(...)` returns a recordset with only itself to prefetch, while `self.with_prefetch()` returns a recordset identical to self except for the prefetching, which is initialized to `self`. This should be enough to "control" the prefetching nightmare caused by the initial code.

Reply:
In fact `self.prefetch` in not even necessary, since no other field is read on it.

Reply:
I agree with the above comment. Since you are modifying these lines anyway, you should use better variable names. `p` and `p2` are confusing since they refer to a different object.